server/graphql/programs/src/mutations/contact_trace/update.rs

```diff
@@ -0,0 +1,125 @@
+use async_graphql::*;
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use repository::{contact_trace::ContactTraceFilter, StringFilter};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    programs::contact_trace::upsert::{UpsertContactTrace, UpsertContactTraceError},
+};
+
+use crate::queries::contact_trace::ContactTraceNode;
+
+#[derive(InputObject)]
+pub struct UpdateContactTraceInput {
+    /// The patient ID the contact belongs to
+    pub patient_id: String,
+    /// The contact trace document type
+    pub r#type: String,
+    /// Contact trace document data
```

Comment:
Could there be different types for different programs ? And I guess program id would come from document registry context for the type ?

Reply:
Yes there could be multiple contact trace types for a single program (theoretically) and yes then you get the program if through the registry.